net: zoap: advance block context by checking M bit from block option

rbtchc opened the pull request:
According to [RFC7959  page 30](https://tools.ietf.org/html/rfc7959#page-30), "The end of a block-wise transfer is
governed by the M bits in the Block options, _not_ by exhausting the
size estimates exchanges."

Therefore, we should check the M bit instead of total size (which
is not always available, too)

Signed-off-by: Robert Chou <robert.ch.chou@acer.com>

rbtchc commented:
Not sure why sanitycheck is failed. Doesn't seem to relate to my modification

pfalcon commented:
@rbtchc : Ack, random qemu failure. Can you re-push this PR? The easiest way to do that is (in the branch of this PR):

~~~
git rebase --ignore-date HEAD^
git push origin HEAD -f
~~~

rbtchc committed:
net: zoap: advance block context by checking M bit from block option

According to RFC7959 page 30, "The end of a block-wise transfer is
governed by the M bits in the Block options, _not_ by exhausting the
size estimates exchanges."

Therefore, we should check the M bit instead of total size (which
is not always available, too)

Signed-off-by: Robert Chou <robert.ch.chou@acer.com>

rveerama1 reviewed (approved):
ACK.
